[No squash] Improve hand override documentation and add special items documentation (#15958)
luanti-org/luanti: doc/lua_api.md

* `craftresult`: list containing the crafted output
* `hand`: list containing an override for the empty hand
    * Is not created automatically, use `InvRef:set_size`
    * Players use the first item in this list as their hand

>     * It behaves like the default hand `""` got overridden for a single player

Reviewer: Is the "" special item even documented?

Author: Hm, most games override the `""` item, but I haven't found a documentation yet, I guess there is none.

Reviewer: I suggest documenting that as well, otherwise you're referring to non-existent documentation here

Author: Now this PR also adds documentation for the "special items", since it would not make sense to only document one of them,

* `craftresult`: list containing the crafted output
* `hand`: list containing an override for the empty hand
    * Is not created automatically, use `InvRef:set_size`
    * Players use the first item in this list as their hand

>     * It behaves like the default hand `""` got overridden for a single player

Reviewer 2: ```suggestion
    * It behaves as if the default hand `""` has been overridden for this specific player
```
